crates/rust-analyzer/src/flycheck.rs

```diff
@@ -396,6 +400,10 @@ impl FlycheckActor {
                     None => cmd.arg("--workspace"),
                 };
 
+                if let Some(tgt) = bin_target {
+                    cmd.arg("--bin").arg(tgt);
```

Comment:
We probably should also pass the package here (non-optional). Two packages in a workspace can have the a binary with the same name. In that case, Cargo checks both of them (tested it).

Reply:
I think its fine to support either, the package is passed above if a package was set (which should be usually the case)

Reply:
Not for Rustlings. But if we extend this to tests instead of only bins, a workspace could have multiple packages with the test `integration_tests` for example. I don't see why rust-analyzer should check all of them if only one changes.

Reply:
Fair, I'd still leave the logic as is though and just change the restart callers as this invocation isn't technically invalid.